notebooks/passenger_vehicle_co2_lca.py

```diff
@@ -0,0 +1,322 @@
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+# ## Grundläggande antaganden
+# Grundläggande antaganden kommer från Volvo Cars LCA-rapporter, som återfinns här:
+#
+# https://www.volvocars.com/mt/news/sustainability/transparency-in-action-heres-our-ex90-carbon-footprint-report/
+#
+# Siffrorna för CO2 per liter bensin kan justeras uppåt något ytterligare enligt denna källa:
+#
+# https://innovationorigins.com/en/producing-gasoline-and-diesel-emits-more-co2-than-we-thought/
+#
+# I denna analys har dock datan från Volvo använts.
+
+# Grundläggande antaganden
+car_manufacturing_cost = 8  # 8 kg CO2/kg car
+battery_manufacturing_cost = 77  # 77 kg CO2/kWh battery (average between 56 & 98, from EX90 and EX30, respectively.)
+kgC02_per_kWh_europe = 0.45  # 0.45 kg CO2/kWh electricity (World mix)
+kgC02_per_litre = 3  # 3kg C02/litre (extraction, production & combustion)
+
+# ## Definiera bilar
+# För att kunna göra jämförelsen behöver vi välja vilka biltyper vi vill titta på. Följande tre typexempel har tagits fram:
+#
+# ##### Version 1
+# |Bil | Vikt exkl batteri (kg) | Batteri (kWh) | Räckvidd el (km) | Energiförbrukning| Användande|
+# |---|---|---|---|---|---|
+# |A: Liten bensinbil | 1500 | - | - | 0.5 liter bensin/mil |100%|
+# |B: Stor elbil | 2200 | 125 | 500 | 0.25 kWh/km |100%|
+# |C: Liten elbil | 1500 | 50 | 330 | 0.15 kWh/km |100%|
+# |D: Cykel | 10 | 0 | 0 | 0kWh/km |100%|
+# |E: Elcykel | 15 |0.5 | 20 | 0.005 kWh/km |100%|
+#
+
+cars = {
+    "A: Liten bensinbil": [1500, 0, 0, 0.05, 1],
+    "B: Stor elbil": [2200, 125, 500, 0.25, 1],
+    "C: Liten elbil": [1500, 50, 330, 0.15, 1],
+    "D: Cykel": [10, 0, 0, 0, 1],
+    "E: Elcykel": [15, 0.5, 20, 0.005, 1],
+}
+print(cars)
+
+
+# Definiera funktion som ger CO2 över tid
+def co2analysis(cars, driven_distance, co2debt):
+    """CO2 accumulator function."""
+    car_data = []
+    for key, value in cars.items():
+        C02_data_constant = (
+            co2debt
+            + car_manufacturing_cost * value[0]
+            + battery_manufacturing_cost * value[1]
+        )
+        if value[1] == 0:
+            C02_data = (
+                C02_data_constant
+                + kgC02_per_litre
+                * value[3]
+                * (driven_distance - driven_distance[0])
+                * value[4]
+            )
+        else:
+            C02_data = (
+                C02_data_constant
+                + kgC02_per_kWh_europe
+                * value[3]
+                * (driven_distance - driven_distance[0])
+                * value[4]
+            )
+        car_data.append([key, driven_distance, C02_data / 1000])
+
+    return car_data
+
+
+driven_distance = np.linspace(0, 300000, 200)
```

Comment:
300000 magic number

Reply:
Yes. Should be defined at the top as an input parameter. Got it from a quick googling.